src/middlewares/constructJrd.ts

```diff
@@ -0,0 +1,35 @@
+import { NextFunction, Request, Response } from 'express'
+
+// eslint-disable-next-line node/file-extension-in-import -- typescript needs .json extension
+import * as discoveryLinks from '../discoveryLinks.json'
+import { ParseError, ParseErrorType } from '../utils/errors'
+
+/**
+ * Constructs a PayID Discovery JRD from a PayID.
+ *
+ * @param req - Contains a PayID as a query parameter.
+ * @param res - Stores the JRD to be returned to the client.
+ * @param next - Passes req/res to next middleware.
+ * @returns A Promise resolving to nothing.
+ * @throws ParseError if the PayID is missing from the request parameters.
+ */
+export default async function constructJrd(
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): Promise<void> {
+  const payId = req.query.resource
+  if (!payId || Array.isArray(payId) || typeof payId !== 'string') {
```

Comment:
Not sure what the array check adds here. Am I missing something?

```suggestion
  if (!payId || typeof payId !== 'string') {
```

Reply:
I think `req.query.resource` is of type `string | string[] | ParsedQs | ParsedQs[]`, so in order to correctly cast it to `string`, I think we needed that check

Reply:
Got it. I'm reading over the WebFinger RFC now, and I didn't think through how this worked.

So basically you make a `GET example.com/.well-known/webfinger?resource=hbergren$example.com` request to get the discovery endpoints for my PayID, correct?

And because `resource` is a query parameter, it can either be a string or an array.

Can you add a comment to that effect above this line to aid reader understanding?

Reply:
Yep you got it.  Will do